credoai/modules/model_assessments/fairness_nlp.py

```diff
@@ -203,9 +202,21 @@ def __init__(self,
             self.use_encoder = hub.load("https://tfhub.dev/google/universal-sentence-encoder/4")
 
     def prepare_results(self):
-        pass
+        if self.raw_results is not None:
+            # Calcualte statistics across subgroups
+            results = self.raw_results[['subgroup','toxicity']].groupby('subgroup', as_index=False).mean()
+            results.loc[len(results)] = ['overall', self.raw_results['toxicity'].mean()] 
+            results['protected_attribute'] = self.raw_results.iloc[0]['protected_attribute']
+            results.rename(columns={'toxicity':'value'}, inplace=True)
+            results['metric'] = 'toxicity'
+            results = results[['metric', 'subgroup', 'protected_attribute', 'value']]
+            return results
+        else:
+            raise Exception(
```

Comment:
If it isn't run, why not just run it, then prepare results? I.e., make the if statement run if not run, then prepare.

Reply:
I actually thought about it and ended up checking how sklearn handles this situation: they do not auto `fit` when `predict` is called before `fit` is done. The error message I have here is also copied from there. I think their logic is to make sure the user 'fit'/'run' with the appropriate parameters (be aware of that).